atom/common/promise_util.h

```diff
@@ -16,16 +16,30 @@ namespace atom {
 
 namespace util {
 
-class Promise : public base::RefCounted<Promise> {
+// A wrapper around the v8::Promise.
+//
+// This is a move-only type that should always be `std::move`d when passed to
+// callbacks, and it should be destroyed on the same thread of creation.
```

Comment:
Can you add a [SequenceChecker](https://cs.chromium.org/chromium/src/base/sequence_checker.h) macro to assert this criteria ?

Reply:
I just tried it and unfortunately it does not support moving, I'll see if I can upstream a change to add move constructor for SequenceChecker.